static/js/components/SpinnerButton_test.js

```diff
@@ -71,11 +78,64 @@ describe("SpinnerButton", () => {
       spinning={true}
       onClick={sandbox.stub()}
       component="button"
-    />);
+    >
+      text
+    </SpinnerButton>);
+    let buttonProps = wrapper.find("button").props();
+
+    assert.equal(buttonProps.className, undefined);
+    assert.isTrue(buttonProps.disabled);
+    assert.equal(buttonProps.onClick, undefined);
+    assert.equal("text", wrapper.find("button").text());
+  });
+
+  it('sets recentlyClicked to true when the button is clicked', () => {
+    let onClick = sandbox.stub();
+    let wrapper = shallow(
+      <SpinnerButton
+        component="button"
+        onClick={onClick}
+        spinning={false}
+      />
+    );
+    assert.isFalse(wrapper.state().recentlyClicked);
     let buttonProps = wrapper.find("button").props();
+    buttonProps.onClick("args");
+    assert.isTrue(onClick.calledWith("args"));
+    assert.isTrue(wrapper.state().recentlyClicked);
+  });
 
+  it('is disabled if spinning is true but recentlyClicked is false', () => {
```

Comment:
maybe it would be better to title this test `'does not show the spinner if spinning is true but recentlyClicked is false'`.